Doc/reference/datamodel.rst

```diff
@@ -1013,6 +1013,36 @@ Internal types
       If a code object represents a function, the first item in :attr:`co_consts` is
       the documentation string of the function, or ``None`` if undefined.
 
+      .. method:: codeobject.co_positions()
+
+         Returns an iterable over the source code positions of each bytecode
+         instruction in the code object.
+
+         The iterator returns tuples containing the ``(start_line, end_line,
+         start_column, end_column)``. The *i-th* tuple corresponds to the
+         position of the source code that compiled to the *i-th* instruction.
+         Column information is 0-indexed utf-8 byte offsets on the given source
+         line.
+
+         This positional information can be missing. A non-exhaustive lists of
+         cases where this may happen:
+
+         - Running the interpreter with ``-X no_debug_ranges``.
+         - Loading a pyc file compiled while using ``-X no_debug_ranges``.
+         - Position tuples corresponding to artificial instructions.
+         - Source code lines past the 255th column (A limitation of columns
```

Comment:
This is still an implementation detail, this case should be 'is not available'

Reply:
Do you want to just omit this line? Batuhan had a point that users may be confused if they suddenly start getting `None` for long lines. I rephrased it to be more generic "implementation specific limitations" and indicated that the CPython one is only currently that way.

Is that better or should we just remove the line altogether?

Reply:
We can rephrase the it as only for CPython (and maybe we should), but it is important to mention about limitations (and availability) since tools who are going to use these APIs need to know them in order to process this information. And if we change the ranges in future, that should also be documented so having an entry point is good about that too.

Reply:
> but it is important to mention about limitations (and availability) since tools who are going to use these APIs need to know them in order to process this information.

That is fine, what is not fine is to document implementation details. We need to say that the availability could be restricted and that some times is not possible to get the information, but we should not mention that this is due to an implementation detail. Just say "in some occasions is not possible to gather the information and we return None".

Otherwise keeping this in sync with the implementation and every possible case when is missing is going to be just a big pain

Reply:
I still prefer to drop the "For example, CPython currently does not store column numbers larger than 255)", but I agree that the sentence as it is currently is better.

Reply:
Removed the 255 limit mention.